Sources/Core/PathExtensions.swift

```diff
@@ -11,7 +11,14 @@ extension Path {
     /// - `../a/b` simplifies to `../a/b`
     /// - `a/../../c` simplifies to `../c`
     public func simplifyingParentDirectoryReferences() -> Path {
-        normalize().components.reduce(Path(), +)
+        if !string.contains("..") { // Skip simplifying if its already simple
```

Comment:
Do we need to look for `../` instead?

Reply:
Sorry for late response. Wouldn't it be like `if !(string.contains("/..") || string.contains("../"))` if we want it to match real parent references only?
I preferred to keep it simple than perfect because false positives like `./file..name/` or `...` are edge cases and won't be big problems IMO, while false negatives can break things.
I guess we could possibly rewrite the complete normalization by ourselves to get both better performance and correctness, as a future task maybe.